app/policies/admin_profile_policy.rb

```diff
@@ -1,10 +1,28 @@
 # frozen_string_literal: true
 
 class AdminProfilePolicy < ApplicationPolicy
+  def show?
+    admin_only
+  end
+
+  def index?
+    admin_only
+  end
+
+  def destroy?
+    admin_only && @record.id != user.id
+  end
```

Comment:
So the admin cannot delete themselves?

Reply:
Exactly - both UI and policy enforces that plus there is a request unit test to cover the case